frontend/src/lib/DestinationForm.svelte

```diff
@@ -0,0 +1,120 @@
+<script>
+    import { redirect } from "@sveltejs/kit";
+    import DestinationBoolParam from "$lib/DestinationBoolParam.svelte";
+    import DestinationNumberParam from "$lib/DestinationNumberParam.svelte";
+    import DestinationPasswordParam from "$lib/DestinationPasswordParam.svelte";
+    import DestinationTextareaParam from "$lib/DestinationTextareaParam.svelte";
+    import DestinationTextParam from "$lib/DestinationTextParam.svelte";
+
+    async function getDestinationParameters(connectionType) {
+        const resource = `/api/destinations/params/${connectionType}`;
+        const options = {};
+        const response = await fetch(resource, options);
+        const parameters = await response.json();
+
+        return parameters;
+    }
+
+    async function emptyParams() {
+        return {
+            form_fields: [],
+        };
+    }
+
+    function getFormValue(form, name) {
+        return form.querySelector(`[name="${name}"]`).value;
+    }
+
+    async function saveNewDestination() {
+        const form = this;
+        const payload = {
+            type: connectionType,
+            name: getFormValue(form, "name"),
+            settings: {}
+        };
+
+        const parameters = await connectionParamsResponse;
+
+        for (const { name } of parameters.form_fields) {
+            payload.settings[name] = getFormValue(form, name);
+        }
+
+        const resource = "/api/destinations";
+        const body = JSON.stringify(payload);
+        const options = {
+            method: "POST",
+            headers: {
+                "content-type": "application/json",
+                "content-length": body.length,
+            },
+            body
+        };
+
+        const response = await fetch(resource, options);
+
+        if (response.ok) {
```

Comment:
The code throws an error upon success. 

![Screenshot 2024-05-09 at 7 09 14 PM](https://github.com/scratchdata/scratchdata/assets/620236/8f628dc9-8085-461d-9acb-fe698ae678b8)

Reply:
I've not used `redirect` before. I'll dig deeper.

Reply:
Replaced `redirect` call with `goto` utility. Seems solid.